brush-core/src/builtins/mapfile.rs

```diff
@@ -93,46 +80,71 @@ impl builtins::Command for MapFileCommand {
 }
 
 impl MapFileCommand {
-    fn read_entries(
+    async fn read_entries(
         &self,
-        mut input_file: openfiles::OpenFile,
+        input_file: openfiles::OpenFile,
+        context: &mut commands::ExecutionContext<'_>,
     ) -> Result<variables::ArrayLiteral, error::Error> {
-        let mut entries = vec![];
-
         let orig_term_attr = setup_terminal_settings(&input_file)?;
 
-        let mut current_entry = String::new();
-        let mut buffer: [u8; 1] = [0; 1]; // 1-byte buffer
+        let mut entries = vec![];
+        let mut line = vec![];
+        let mut idx = self.skip_count;
+        let mut read_count = 0;
+        let delimiter = self.delimiter.chars().next().unwrap_or('\n') as u8;
+
+        let mut reader = input_file.clone();
```

Comment:
Now that the `BufReader` has been removed, I think you should be able to remove the `.clone()`. For that matter, it will probably be simpler to mark `input_file` as `mut` and remove the `reader` variable.